### PR TITLE
Bridge client

### PR DESCRIPTION
**Pull request by markrypto**
This is the Windows bridge for KeepKey

### CODE REVIEW

File: keepkeylib/transport.py [hunk 1]
```diff
@@ -68,6 +74,12 @@ def write(self, msg):
         header = struct.pack(">HL", mapping.get_type(msg), len(ser))
         self._write(b"##" + header + ser, msg)
 
+    def bridgeWrite(self, msg):
+        """
+        Write mesage to tansport.  msg should be a member of a valid `protobuf class <https://developers.google.com/protocol-buffers/docs/pythontutorial>`_ with a SerializeToString() method.
```
Comment (reidrankin) [inline keepkeylib/transport.py:79]:
```suggestion
        Write message to transport. msg should be a member of a valid `protobuf class <https://developers.google.com/protocol-buffers/docs/pythontutorial>`_ with a SerializeToString() method.
```

File: README.rst [hunk 2]
```diff
@@ -118,3 +117,49 @@ Release Process
   * sudo python3 setup.py sdist bdist_wheel bdist_egg
 * Upload the release
   * sudo python3 -m twine upload dist/* -s --sign-with gpg2
+
+KeepKey Bridge
+==============
+The KeepKey Bridge is a standalone TCP-to-webusb bridge the KeepKey. It runs a python-keepkey client
+based process that allows a local host-based process to communicate with the KeepKey wallet, thus
```
Comment (reidrankin) [inline README.rst:124]:
```suggestion
based process that allows a localhost-based process to communicate with the KeepKey wallet, thus
```

File: wait-serv.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from waitress import serve
+import kkbridge
+serve(kkbridge.create_app(), host='0.0.0.0', port=1646)
```
Comment (reidrankin) [inline wait-serv.py:3]: pretty sure this is world-accessible?
```suggestion
serve(kkbridge.create_app(), host='127.0.0.1', port=1646)
```

File: kkbridge.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            kkClient = initDevice()
+        else:
+            pass
+
+        if (kkClient == None):
+            data = "No KeepKey found"
+            return Response(str(data), status=404, mimetype='application/json')
+
+        try:
+            ping = kkClient.call(messages.Ping(message='Duck, a bridge!', button_protection = True))
```
Comment (reidrankin) [inline kkbridge.py:77]: lol.

File: keepkeylib/transport.py [hunk 5]
```diff
@@ -93,6 +105,18 @@ def read_blocking(self):
 
         return self._parse_message(data)
 
+    def bridge_read_blocking(self):
+        """
+        blocks untill data is available to be read.
```
Comment (reidrankin) [inline keepkeylib/transport.py:110]:
```suggestion
        blocks until data is available to be read.
```

File: README.rst [hunk 6]
```diff
@@ -118,3 +117,49 @@ Release Process
...
+    ``windows/KeepKeyBridge.iss``
+
+This will produce an executable install app
+
+    ``windows/Output/kkbsetup.exe``
+
+Running the KeepKey Bridge
+--------------------------
+When running the KeepKey Bridge, a blank cmd window with the title "KepKey Bridge" will be visible. 
+To stop the bridge, simply close the cmd window.
```
Comment (mvpratt) [inline README.rst:165]: seems like this should this be combined with the earlier "Running the KeepKey Bridge section"

Comment (markrypto) [reply README.rst]: Ha, this is why I need an editor. Fixed.

File: kkbridge.py (hunk 4 above)
Comment (markrypto) [reply kkbridge.py:77]: 🦆

**Review by mvpratt — APPROVED** [review]

**Pull request merged**